test/legacy/plugin.js

```diff
@@ -154,28 +158,6 @@ describe('promise plugin', function() {
   });
 });
 
-describe('custom plugin', function() {
```

Comment:
This is now an invalid test. All custom plugins must be a class that inherits from `BasePlugin`.

Reply:
As discussed I think we should deprecate the old style custom plugin a bit more gently. I'm happy for that second path to be done in another PR if it is easier.

Reply:
Okay, this makes sense. I'll put together a PR to reinstate this test and handle legacy plugins.